space-cli/cmd/modules/auth/commands.go

```diff
@@ -8,15 +8,28 @@ import (
 
 // GetSubCommands is the list of commands the auth module exposes
 func GetSubCommands() []*cobra.Command {
-	var getAuthProvider = &cobra.Command{
-		Use:  "auth-provider",
-		RunE: actionGetAuthProviders,
-	}
 	var getAuthProviders = &cobra.Command{
-		Use:  "auth-providers",
-		RunE: actionGetAuthProviders,
+		Use:     "auth-providers",
+		Aliases: []string{"auth-provider"},
+		RunE:    actionGetAuthProviders,
+		ValidArgsFunction: func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+			project, check := utils.GetProjectID()
+			if !check {
+				_ = utils.LogError("Project not specified in flag", nil)
+				return nil, cobra.ShellCompDirectiveDefault
+			}
```

Comment:
use log debug, everywhere. user should not get this error when tab tab is pressed